lib/grape/router.rb

```diff
@@ -46,7 +46,7 @@ def compile!
     end
 
     def append(route)
-      map[route.request_method.to_s.upcase] << route
+      map[Grape::Http::Headers.find_supported_method(route.request_method.to_s)] << route
```

Comment:
The `request_method` method comes from [this place](https://github.com/ruby-grape/grape/blob/master/lib/grape/router/route.rb#L72), so `upcase` isn't needed here :slightly_smiling_face: